galley/pkg/runtime/projections/serviceentry/handler.go

```diff
@@ -70,7 +69,6 @@ type Handler struct {
 	// The version number for the current State of the object. Every time mcpResources or versions change,
 	// the version number also change
 	version      int64
-	mcpLock      sync.Mutex
```

Comment:
FYI this lock was never really needed since we always access from the same thread. Removing for now.